diff --git a/lib/notarytool.js b/lib/notarytool.js
index 76d3556e3b4ed304ef29a3166cb9554edf6b83b8..6c4be1ec5a6c0c9b9a8f1f3920e89d1aaac17d9e 100644
--- a/lib/notarytool.js
+++ b/lib/notarytool.js
@@ -110,6 +110,7 @@ function notarizeAndWaitForNotaryTool(opts) {
                 'json',
             ];
             const result = yield (0, spawn_1.spawn)('xcrun', notarizeArgs);
+            d('notarization result', result.output);
             const parsed = JSON.parse(result.output.trim());
             if (result.code !== 0 || !parsed.status || parsed.status !== 'Accepted') {
                 try {
