diff --git a/dist/index.js b/dist/index.js
index 6268f6bd333fb8949b63368050f1400eab7e7a3a..29d42354135a60760b6750aad209d9b8bd891253 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -214,7 +214,7 @@ class TerserPlugin {
               // eslint-disable-next-line global-require
               'terser-webpack-plugin': require('../package.json').version,
               'terser-webpack-plugin-options': this.options,
-              hash: _crypto.default.createHash('md4').update(input).digest('hex')
+              hash: _crypto.default.createHash('md5').update(input).digest('hex')
             };
             task.cacheKeys = this.options.cacheKeys(defaultCacheKeys, file);
           }