diff --git a/node_modules/webpack/lib/HashedModuleIdsPlugin.js b/node_modules/webpack/lib/HashedModuleIdsPlugin.js
index 7a860f7..ef7df31 100644
--- a/node_modules/webpack/lib/HashedModuleIdsPlugin.js
+++ b/node_modules/webpack/lib/HashedModuleIdsPlugin.js
@@ -23,7 +23,7 @@ class HashedModuleIdsPlugin {
 		this.options = Object.assign(
 			{
 				context: null,
-				hashFunction: "md4",
+				hashFunction: "md5",
 				hashDigest: "base64",
 				hashDigestLength: 4
 			},
diff --git a/node_modules/webpack/lib/WebpackOptionsDefaulter.js b/node_modules/webpack/lib/WebpackOptionsDefaulter.js
index f0e552f..060d5fc 100644
--- a/node_modules/webpack/lib/WebpackOptionsDefaulter.js
+++ b/node_modules/webpack/lib/WebpackOptionsDefaulter.js
@@ -159,7 +159,7 @@ class WebpackOptionsDefaulter extends OptionsDefaulter {
 		this.set("output.crossOriginLoading", false);
 		this.set("output.jsonpScriptType", false);
 		this.set("output.chunkLoadTimeout", 120000);
-		this.set("output.hashFunction", "md4");
+		this.set("output.hashFunction", "md5");
 		this.set("output.hashDigest", "hex");
 		this.set("output.hashDigestLength", 20);
 		this.set("output.devtoolLineToLine", false);
diff --git a/node_modules/webpack/lib/optimize/SplitChunksPlugin.js b/node_modules/webpack/lib/optimize/SplitChunksPlugin.js
index e7d560b..10e8383 100644
--- a/node_modules/webpack/lib/optimize/SplitChunksPlugin.js
+++ b/node_modules/webpack/lib/optimize/SplitChunksPlugin.js
@@ -22,7 +22,7 @@ const deterministicGroupingForModules = /** @type {function(DeterministicGroupin
 
 const hashFilename = name => {
 	return crypto
-		.createHash("md4")
+		.createHash("md5")
 		.update(name)
 		.digest("hex")
 		.slice(0, 8);
diff --git a/node_modules/webpack/lib/util/createHash.js b/node_modules/webpack/lib/util/createHash.js
index 64de510..1d59c1b 100644
--- a/node_modules/webpack/lib/util/createHash.js
+++ b/node_modules/webpack/lib/util/createHash.js
@@ -124,6 +124,11 @@ class DebugHash extends Hash {
  * @returns {Hash} the hash
  */
 module.exports = algorithm => {
+  // arg mutation... lovely xD
+  if (algorithm === "md4") {
+    algorithm = "md5"
+  }
+  
 	if (typeof algorithm === "function") {
 		return new BulkUpdateDecorator(new algorithm());
 	}
